### PR TITLE
Implemented step_rate_generator for piecewise constant rate input

### PR DESCRIPTION
**Pull request by janhahne**
This PR implements a feature requested on the NEST user mailing list. The `step_rate_generator` is a modified version of the `step_current_generator` that provides the possibility to send piecewise constant input to rate units.

The original code for the device was written by Sandra Nestler (master student in Juelich) and @ddahmen.

### CODE REVIEW

**Comment by janhahne** [comment]
@jougs @jhnnsnk Any chance of a review in the next weeks? This should hopefully be not that time-consuming (as it is basically a adjusted copy of the already reviewed `step_current_generator`).

File: models/step_rate_generator.h [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+The rate_generator provides a piecewise constant rate input to the
+connected rate unit(s). Please note that this input is handled in the same
+way as input from any other rate unit, i.e. it is processed by the input
+function of the receiving rate unit. The amplitude of the rate is changed
+at the specified times. The unit of the rate is Hz.
+
+Parameters:
+
+The following parameters can be set in the status dictionary:
+amplitude_times   list of doubles - Times at which current changes in ms
```
Comment (jougs) [inline models/step_rate_generator.h]: Shoud be "rate" instead of "current"

File: models/step_rate_generator.h [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+amplitude_values  list of doubles - Amplitudes of step current current in
+                                    pA
```
Comment (jougs) [inline models/step_rate_generator.h]: The unit of the "rates" is most likely not pA, nor is "current current" a good explanation for them

**Comment by janhahne** [comment]
@jougs Thanks for the review and sorry about the missing changes in the parameter description. I have now double-checked the documentation and fixed the parameter description with 07c2b5c. Please have another look.

**Pull request merged**